contents/blog/product-engineer-vs-software-engineer.md

```diff
@@ -0,0 +1,66 @@
+
+Many software engineers have less say in what they work on. They have flexibility with implementation, but they aren’t making product decisions. Their team helps them with that, meaning they can focus on the technical implementation.
+
+Product engineers are less focused on implementation and focus more on solving the users’ problems. In this way, they are pragmatic. They are willing to build fast, iterate, or even from scratch if needed.
+
+Software engineers are more idealistic. They look for the best solution to the problem in front of them, focus on best practices, and build off prior work. Security, risk, and scaling weigh heavier in their minds.
+
+## How does each spend their time?
+
+Product engineers spend their time in a variety of ways. A majority of their time is still spent coding, but they have responsibilities outside of coding than software engineers.
+
+Product engineers also spend their time talking to customers, digging into usage data, and researching the competitive landscape. They test other products, build prototypes, and brainstorm experiments to make their product better.
+
+They spend time communicating with users, their team, and other members of the company. They communicate what they learned in research, why their work relates to that research, and how to prioritize what to work on next. They own the product roadmap, and this comes with a responsibility to communicate it.
+
+Software engineers focus on writing code, testing it, and maintaining it. They  look for ways to optimize code, improve scalability, and solve bugs. They are deeply focused on specific areas of technology, whether that is databases, data pipelines, backend APIs, or client-side app frameworks. This means researching and reading documentation, updates, and code.
+
+## What do their teams and companies look like?
+
+Product engineers are often a part of smaller teams, where they must do more themselves. Startups, and earlier-stage companies that require people to do more, are often looking for product engineers (even if they don’t say it). They need people who can own large decisions and ship quickly. Product engineers fit this need.
+
+Software engineers are part of larger teams, usually spread out across multiple teams. They have a team to support them, so they can specialize. Their team might include designers, product managers, and marketers. This allows them to think less about if what they are building is right, and more about the implementation.
+
+At the end of the day, it’s up to the company and team to decide what roles they need. There are always debates about what the right structure and right roles are. Product engineers and software engineers both have their place in the tech world. It all comes down to the goals of the team and the needs of the customers.
```

Comment:
I think we should refocus this section on "Why do companies want product engineers?" and go a tiny bit deeper here. 

For example, while their most common at early-stage companies, how do product engineers fit into larger orgs – if they do?

I think we could make this a bit topical by maybe referencing Elon's adventures at Twitter here. Explicit or not, what he's seeking is essentially product engineers who can ideate and execute ideas. (my interpretation, feel free to disagree).

The theme is small companies need T-shaped engineers and large companies want them to make them more agile and responsive to users. (again, my interpretation!)

Reply:
Got it, added a bit about this and added the example of Elon's adventures.